packages/lambdas/src/fhir-to-medical-record.ts

```diff
@@ -96,209 +97,1282 @@ export const handler = Sentry.AWSLambda.wrapHandler(
   }
 );
 
-async function getSignedUrl(fileName: string) {
-  return coreGetSignedUrl({ fileName, bucketName, awsRegion: region });
+const jsonToHtml = (json: string): string => {
+  const fhirBundle: Bundle = JSON.parse(json);
+
+  const findPatient = fhirBundle.entry?.find(entry => {
+    return entry?.resource?.resourceType === "Patient";
+  });
+
+  const patient = findPatient ? (findPatient.resource as Patient) : undefined;
```

Comment:
general comment/suggestion:

pattern set here is we're looking through all resources for each resource type we want to extract, and then looping through again to cast to the type

we could save a lot of loops by just looping over all resources once at the top of the function, and extracting to arrays (or maps for quick ID lookup), once

for example:
```
  const progressNotes: DiagnosticReport[] = [];
  const conditions: Condition[] = [];
  if (fhirBundle.entry) {
    for (const entry of fhirBundle.entry) {
      if (entry.resource?.resourceType === "DiagnosticReport") {
           progressNotes.push(entry.resource);
      } else if (entry.resource?.resourceType === "Condition") {
           conditions.push(entry.resource);
    }
  }
```

... and etc - this way we'd only need to loop over all resources once